src/main/java/com/google/sps/servlets/TripServlet.java

```diff
@@ -61,13 +86,74 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     eventDoGet(response);
   }
 
+  /**
+   * Get user input.
+   * Generate directionsRequest from user input and parse optimized route.
+   * Create and store events in Datastore.
+   */
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) 
       throws IOException {
     response.setContentType("application/json;");
 
+    Enumeration<String> params = request.getParameterNames();
+    String tripName = request.getParameter("inputTripName");
+    String origin = request.getParameter("inputDestination");
+    String startDate = request.getParameter("inputDayOfTravel");
+
+    // Print user input for now
+    response.getWriter().println("startDate: " + startDate);
+    response.getWriter().println("tripName: " + tripName);
+    response.getWriter().println("origin: " + origin);
+
+    // Save POIs to an array
+    List<String> pois = new ArrayList<>(); 
+    while (params.hasMoreElements()) {
+      String paramName = params.nextElement();
+      if (paramName.contains("poi")) {
+        String newPOI = request.getParameter(paramName);
+        pois.add(newPOI);
+      }
+    }
+    String[] poiStrings = new String[pois.size()]; 
+    poiStrings = pois.toArray(poiStrings); 
+
+    response.getWriter().println("original POI order: " + pois.toString());
+
+    //do post for maps
+    mapDoPost(request, response, origin, poiStrings);
+
     // do post for events
-    eventDoPost(request, response); 
+    eventDoPost(request, response, startDate, poiStrings); 
+  }
+
+  /**
+   * Make servlet cleaner
+   * Generate directionsRequest from user input and parse optimized route.
+   */
+  private void mapDoPost(HttpServletRequest request, HttpServletResponse response, 
+                        String origin, String[] pois) 
+      throws IOException {
+    GeoApiContext distCalcer = new GeoApiContext.Builder()
+      .apiKey(Config.API_KEY)
+      .build();
+
+    DirectionsApiRequest directionsRequest = generateDirectionsRequest(origin, pois, distCalcer);
+
+    // Calculate route and save travelTimes and waypointOrder to two ArrayLists.
+    try {
+      DirectionsResult dirResult = directionsRequest.await();
+
+      List<Integer> travelTimes = getTravelTimes(dirResult);
```

Comment:
If you could have this function return travelTimes that would be super helpful for me / necessary to when I create the events.